plugin.py

```diff
@@ -112,10 +124,33 @@ def run() -> None:
 
         sublime.set_timeout_async(run)
 
-    def m_dart_textDocument_publishClosingLabels(self, params: Any) -> None:
-        # TODO: Implement me.
-        pass
 
+
+    def apply_closing_labels(self, view, labels):
+        for label in labels:
+            final_character_position = label["range"]["end"]
+            if final_character_position["character"] < 1:
+                return
+            row = label["range"]["end"]["line"]
+            point = view.line(view.text_point(row, 0)).end()
+            view.add_phantom(closing_labels,
+                                sublime.Region(point, point+1),
+                                build_label(view, label["label"]),
+                                sublime.LAYOUT_INLINE)
```

Comment:
This feels a bit dangerous. When you disable the plugin, are these phantoms removed?

The presentation of these labels is similar to what happened in LSP-metals: https://github.com/scalameta/metals-sublime/pull/36

What we ended up doing there was to stash a `sublime.PhantomSet` inside of a `SessionView`. This guarantees that the lifetime of these phantoms are bound to the lifetime of the `SessionView`. LSP tries to init/del those `SessionView`s in a correct manner.

So try something like this:

```python
for sv in session_buffer.session_views:
     try:
         phantom_set = getattr(sv, "_lsp_dart_labels")
     except AttributeError:
         phantom_set = sublime.PhantomSet(sv.view, self.phantom_key)
         setattr(sv, "_lsp_dart_labels", phantom_set)

     phantom_set.update(self.apply_closing_labels(sv, reversed(params["label"])))
```

Reply:
Interesting! I'll check to make the changes with that method, it might even solves a visual "glitch" I have where when the labels are redrawn, they flicker (erased then drawn must it's not instant)

Reply:
just made the changes with a `PhantomSet`, it provides a better UX as we don't have the flickering as each redraw in addition to the issue you where referring